internal/testing/conformance/run_conformance_test.go

```diff
@@ -0,0 +1,48 @@
+//go:build conformance
+// +build conformance
+
+package conformance
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/consul-api-gateway/internal/testing/e2e"
+	"sigs.k8s.io/e2e-framework/pkg/env"
```

Comment:
So, I'm _somewhat_ indifferent on this front, but personally I'd probably prefer not to mix/match testing frameworks, since they're both rather opinionated in their setup. I'm wondering if maybe first pass we can just work on the assumption that we have a functional `kubernetes` cluster at this stage (with, in the case of `kind`, already has `MetalLB` up and running) and potentially even the gateway controller already installed on the system (which we could make easier down the line if we so desire).

That way this would only involve the _much_ smaller task of rendering out the patched manifests and invoking the underlying conformance suite code.

Much of the setup logic for CI could then just get moved to the yaml manifest and we could keep the `conformance_test.go` parts closer the vanilla upstream. If we wanted to refactor down the line to do more of the setup in `go` that'd be cool too, but I figure that might be able to wait a bit.

Reply:
If we were to do it that way and pull it all out into the CI work, we could just use the kind action Nathan linked and simplify a lot of things. I'm pretty indifferent too, I just thought we wanted it all to be contained in `go test`. Definitely happy to spend some time proving out this other way though.

Reply:
I'm ok without it being _fully_ `go test`able at the get-go, so long as the major patching bits an actual test invocation are done in code at the beginning and `go test`able.

I'm also ok extending some of this work down-the-line to do the same random env spin-up we do in our `e2e` tests while just doing the minimal thing here. I think it'd be fine to hold off on doing the whole go-based `kind` spin-up in this PR, especially as we've talked about changing the way some of our e2e environments get provisioned to be more extensible for cloud systems and not just purely `kind`-based.